### PR TITLE
Resolved 'When we scale the app, newly created replica shows as Replica 1'

### PR DESCRIPTION
**Pull request by madeesha**

### CODE REVIEW

File: modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java [hunk 1]
```diff
@@ -384,7 +384,19 @@ public DeploymentLogs getRuntimeLogs(LogQuery query) throws RuntimeProvisioningE
...
                 int podCounter = 1;
-                for (Pod pod : podList.getItems()) {
+
+                /** Sorting the podList
+                 */
+                List<Pod> unsortedList = podList.getItems();
+                Collections.sort( unsortedList, new Comparator<Pod>() {
+                    //Sort podList by app creation time
+                    @Override
+                    public int compare(Pod a, Pod b) {
```
Comment (Sithumini) [inline modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java]: @madeesha shall we assign meaningful variable names than a,b ?

File: modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java [hunk 2]
```diff
@@ -384,7 +384,19 @@ public DeploymentLogs getRuntimeLogs(LogQuery query) throws RuntimeProvisioningE
         if (podList != null) {
             try {
                 int podCounter = 1;
-                for (Pod pod : podList.getItems()) {
+
+                /** Sorting the podList
+                 */
+                List<Pod> unsortedList = podList.getItems();
+                Collections.sort( unsortedList, new Comparator<Pod>() {
```
Comment (Sithumini) [inline modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java]: @madeesha minor formatting issue

File: modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java [hunk 3]
```diff
@@ -384,7 +384,19 @@ public DeploymentLogs getRuntimeLogs(LogQuery query) throws RuntimeProvisioningE
...
+                    //Sort podList by app creation time
+                    @Override
+                    public int compare(Pod a, Pod b) {
+                        return a.getMetadata().getCreationTimestamp().toString().compareTo(b.getMetadata().getCreationTimestamp().toString());
```
Comment (Sithumini) [inline modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java]: @madeesha formatting issues. this line seems longer than 120 characters.

File: modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java [hunk 4]
```diff
@@ -1111,7 +1124,25 @@ public JSONArray getReplicaInfo() throws RuntimeProvisioningException {
...
+            JSONArray sortedJsonArray = new JSONArray();
+            List<JSONObject> jsonValues = new ArrayList<JSONObject>();
+
+            for (int i = 0; i < jsonArray.length(); i++) {
+                jsonValues.add(jsonArray.getJSONObject(i));    // put all JSON objects to the array list
+            }
+            Collections.sort( jsonValues, new Comparator<JSONObject>() {
+                //Sort by app creation time
+                @Override
+                public int compare(JSONObject a, JSONObject b) {
```
Comment (Sithumini) [inline modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java]: @madeesha use meaningful variable names

File: modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java (hunk 2 above)
Comment (madeesha) [inline modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java]: Fixed

File: modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java (hunk 1 above)
Comment (madeesha) [inline modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java]: fixed

File: modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java (hunk 3 above)
Comment (madeesha) [inline modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java]: fixed

File: modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java (hunk 4 above)
Comment (madeesha) [inline modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java]: fixed

**Comment by amilam** [comment]
Looks good.

**Pull request merged**